Shared/Samples/Show line of sight between geoelements/ShowLineOfSightBetweenGeoelementsView.swift

```diff
@@ -0,0 +1,357 @@
+                sceneSymbol.anchorPosition = .bottom
+                taxiGraphic = Graphic(
+                    geometry: taxiPoint,
+                    symbol: sceneSymbol
+                )
+                let camera = Camera(
+                    lookingAt: observerPoint,
+                    distance: 700.0,
+                    heading: -30.0,
+                    pitch: 45.0,
+                    roll: 0.0
+                )
+                scene.initialViewpoint = Viewpoint(
+                    boundingGeometry: observerPoint,
+                    camera: camera
+                )
+                observerGraphic = Graphic(
+                    geometry: observerPoint,
+                    symbol: observerSymbol
+                )
+                addGraphicsToOverlays()
+                displayLink.isPaused = false
+            } catch {
+                self.error = error
+            }
+        }
+        
+        /// Adds the observer, target, and analysis objects to their respective overlays.
+        private func addGraphicsToOverlays() {
+            if let observer = observerGraphic, let taxi = taxiGraphic {
+                graphicsOverlay.addGraphic(observer)
+                graphicsOverlay.addGraphic(taxi)
+                lineOfSight = GeoElementLineOfSight(
+                    observer: observer,
+                    target: taxi
+                )
+                lineOfSight?.targetOffsetZ = 2
+                if let lineOfSight = lineOfSight {
+                    analysisOverlay.addAnalysis(lineOfSight)
+                }
+            }
+        }
+        
+        /// Creates a display link timer for the image overlay animation.
+        /// - Returns: A new `CADisplayLink` object.
+        private func makeDisplayLink() -> CADisplayLink {
+            // Create new display link.
+            let newDisplayLink = CADisplayLink(
+                target: self,
+                selector: #selector(animateTaxi)
+            )
+            // Set the default frame rate to 60 fps.
+            newDisplayLink.preferredFramesPerSecond = 60
+            newDisplayLink.isPaused = true
+            // Add to main thread common mode run loop, so it is not effected by UI events.
+            newDisplayLink.add(to: .main, forMode: .common)
+            return newDisplayLink
+        }
+        
+        /// Used for deallocating `displayLink` on view disappear.
+        func stopAnimating() {
+            displayLink.invalidate()
+            displayLink = nil
+        }
+        
+        /// Animates the target graphic between a set of points in a loop,
+        /// updating the heading and visibility analysis on each frame.
+        @objc
+        private func animateTaxi() {
+            guard let taxiGraphic = taxiGraphic else { return }
+            // Increment the frame counter
+            frameIndex += 1
+            // Reset frame counter when segment is completed
+            if frameIndex == frameMax {
```

Comment:
No need to change as the points array here is quite small. FYI there is a `CircularIterator` type in other samples that does this logic, if you want to use.